CONTRIBUTING.md

```diff
@@ -281,6 +281,8 @@ If you need to modify or debug the documentation flow, these are the main files:
   src/nvim/eval.lua     =>  runtime/doc/builtin.txt
   src/nvim/options.lua  =>  runtime/doc/options.txt
   ```
+- `./scripts/gendoc_ex_cmd_index.lua`:
+  Generates the index of Ex commands into `runtime/doc/index.txt`, used by `gen_vimdoc.py`.
```

Comment:
BTW, `gen_eval_files.lua` also renders some part of vimdoc (e.g. `options.txt` `vvars.txt`) so we should consolidate them. I now think `gen_eval_files.lua` would be a more appropriate place for `|ex-cmd-index|` than `gen_vimdoc`, so will revert `gen_vimdoc.py`:

- `gen_vimdoc.py` is mainly for C/Lua docstrings -> vimdoc (Function/Class documentation).
- `gen_eval_files.lua` is mainly for the *opposite*, i.e. some source of truth -> Lua meta files (typing and docstring) and vimdoc (index)

Reply:
> `gen_eval_files.lua` is mainly for the _opposite_, i.e. some source of truth -> Lua meta files (typing and docstring) and vimdoc (index)

Looks like `gen_eval_files.lua` already generates vimdoc, so I guess it doesn't matter either way. We should do whatever is less confusing for contributors. 

We want to replace `gen_vimdoc.py` with a treesitter-based Lua script, so it probably makes sense to avoid enhancements to it where possible